queries/nim/injections.scm

```diff
@@ -0,0 +1,82 @@
+; =============================================================================
+; generalized_strings
+
+; regex in generalized_strings
+(generalized_string 
+  function: (identifier) @_string_prefix .
+  (string_content) @injection.content
+  (#set! injection.language "regex")
+  (#any-of? @_string_prefix "re" "rex"))
+
+; format string in generalized_strings
+(generalized_string 
+  function: (identifier) @_string_prefix .
+  (string_content) @injection.content
+  (#set! injection.language "nim_format_string")
+  (#eq? @_string_prefix "fmt"))
+
+; format string in normal strings with & prefix
+(prefix_expression
+  operator: (operator) @_string_prefix .
+  [
+    (interpreted_string_literal (string_content) @injection.content)
+    (long_string_literal (string_content) @injection.content)
+    (raw_string_literal (string_content) @injection.content)
+  ] 
+  (#set! injection.language "nim_format_string")
+  (#eq? @_string_prefix "&"))
+
+; sql in generalized_strings
+; and anyhting you like as long as the function name is the same as the injected language's parser
+(generalized_string 
+  function: (identifier) @injection.language
+  (string_content) @injection.content
+  (#not-any-of? @injection.language "re" "rex" "fmt"))
+
+; =============================================================================
+; asm statement
+
+; there is no ts parser for assembly
+; (assembly_statement (string_literal) @injection.content
+;   (#set! injection.language "assembly"))
+
+; =============================================================================
+; emit pragma
+
+; C / CPP / OBJC / JAVASCRIPT
+; a `#<no whitespace><language name>`
+; has to directly precede the {.emit: "<language code>".} pragma
+
+; eg.:
+; #objc
+; {.emit: "<objc code>".}
+; OR
+; #javascript
+; {.emit: "<javascript code>".}
+
+; normal strings
+((comment (comment_content) @injection.language) 
+  .
+  (pragma_statement
+    (pragma_list
+      (colon_expression
+        left: (identifier) @_emit_keyword (#eq? @_emit_keyword "emit")
+        right: [
+          (interpreted_string_literal (string_content) @injection.content)
+          (long_string_literal (string_content) @injection.content)
+          (raw_string_literal (string_content) @injection.content)
+        ]))))
```

Comment:
This should work.
```suggestion
        right: (_ (string_content) @injection.content)))))
```

Reply:
is that better for performance?

There are other places where I opted for explicit options over using `_` if they are limited.

Reply:
It's less likely to break if nodes are added or removed from the parser, for one thing.

Reply:
doesn't feel like a strong argument for me. everything is likely to break, that's why the parser version is pinned.

more generic can be as bad more specific in terms of breakage.

Eg there could be a new node introduced that contains `(string_content)`

Reply:
> is that better for performance?

I don't know about performance but it's better for readability.

> Eg there could be a new node introduced that contains `(string_content)`

That would still be injectable, no?

Reply:
> doesn't feel like a strong argument for me. everything is likely to break, that's why the parser version is pinned.

Yes, but we still _try_ to update, and the less manual intervention is needed, the better for us and for you (as the language maintainer).

Reply:
> That would still be injectable, no?

Yes, but then it could be injected in the wrong places. And then it would be a silent error. This way we get immediate feedback if something is not working as intended

Reply:
In this case, for the query to be injected in the wrong place, it means there has to be a new node with `(string_content)` that will be placed in the same scenario as this. And most likely that node will also be a string-esque that will need to be injected.

Reply:
Still. All we gain is subjective readability. But I can do it if you all insist.